triangle indicator for dropdowns

Frizi opened the pull request:
### Pull Request Description

Closes #5854 

Switches dropdown activation indicator to a triangle shape, and moved it to the horizontal center of a port.
![image](https://user-images.githubusercontent.com/919491/223765985-ec2175b7-7b44-45fd-88ff-543e8c08538f.png)


### Important Notes

Modified triangle SDF to be exact. That way the grow operation behaves as expected, rounding the corners. Other than that, it produces the same bound shape at 0 distance.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using `./run ide build`.

Frizi committed:
Switch dropdown activation shape to a triangle

Frizi committed:
lint

wdanilo commented on lib/rust/ensogl/core/src/display/shape/primitive/def/primitive.rs:
```diff
@@ -332,11 +332,20 @@ define_sdf_shapes! {
 
     // === Triangle ===
 
+    /// Isosceles Triangle
```
What is this line? Is this the first sentence? If so, its missing a dot.

wdanilo commented on lib/rust/ensogl/core/src/display/shape/primitive/def/primitive.rs:
```diff
@@ -332,11 +332,20 @@ define_sdf_shapes! {
     // === Triangle ===
 
+    /// Isosceles Triangle
+    /// This is an exact SDF. Growing it will result in a triangle with rounded corners.
```
What do you mean by "This is an exact SDF"?

Frizi replied:
By that I mean it's not an approximation that holds true only inside the shape. The SDF function calculates a true distance, which is correct also far away from the shape boundary. That allows grow/shrink to work as expected. I've made a comparsion in shadertoy to illustrate that.

https://www.shadertoy.com/view/msG3W1
|Exact SDF|Bound SDF|
|-|-|
|![image](https://user-images.githubusercontent.com/919491/224285071-901613ed-bd9c-41ce-9e92-dada5dee30a3.png)|![image](https://user-images.githubusercontent.com/919491/224285180-abdcda9b-d650-4e9b-9a40-18e05e41bc51.png)|

wdanilo replied:
Oh, I when reading the comment I did not know that the sentence "This is an exact SDF" is explained by the following sentence! I think it would be clear if you use comma instead of dot in the description then ...

Frizi committed:
adjust comment wording to clear up what exact SDF means.

farmaazon commented:
## QA Report

I found one issue: once I pick some entry in the drop-down widget, the top parts of the drop-down stop reacting to mouse events. E.g. when I create a `filter` node and pick one of two columns, then I cannot select the other one. Checked develop - there is no such issue.


https://user-images.githubusercontent.com/3919101/224345931-1728456c-50d1-4330-87d9-adf0f28c1eab.mp4

Frizi committed:
Merge branch 'develop' into wip/frizi/dropdown-triangles

Frizi commented:
@farmaazon That issue was introduced in #5688 and fixed in https://github.com/enso-org/enso/pull/5774/files#diff-50b0b3ebb72f152c5f3f420e6fc569c6bd2342d068c5eda7805a7ba97006f022R120, which is already merged. So that explain why it's not in develop. Rebasing this PR should get rid of it.


<blockquote><img src="https://avatars.githubusercontent.com/u/919491?s=400&v=4" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/enso-org/enso/pull/5774">Named arguments support in IDE by Frizi · Pull Request #5774 · enso-org/enso</a></strong></div><div>Pull Request Description
Added support for named arguments in IDE.


  
Named arguments are now recognized in node expressions. The function argument placeholders are rendered around series of name...</div></blockquote>

farmaazon commented:
After the merge I see another problem: the labels in drop-downs are not centered vertically. Compare this screenshot with  the clip above

![image](https://user-images.githubusercontent.com/3919101/224365017-6e0a4d7a-5755-4d54-b9cb-9d1998ef2188.png)

But it may be also on develop. If it is, then the PR may be merged, no other issues found.

Frizi commented:
It is the same on latest develop.
![image](https://user-images.githubusercontent.com/919491/224435470-4a178129-3e3b-40a8-a5dd-607962ae9880.png)

mergify[bot] committed:
Merge branch 'develop' into wip/frizi/dropdown-triangles

mergify[bot] committed:
Merge branch 'develop' into wip/frizi/dropdown-triangles

mergify[bot] committed:
Merge branch 'develop' into wip/frizi/dropdown-triangles

mergify[bot] committed:
Merge branch 'develop' into wip/frizi/dropdown-triangles
